File: app/src/main/java/org/tasks/caldav/UpdateCaldavAccountViewModel.java
package org.tasks.caldav;

import org.tasks.ui.CompletableViewModel;

@SuppressWarnings("WeakerAccess")
public class UpdateCaldavAccountViewModel extends CompletableViewModel<String> {
  void updateCaldavAccount(
      CaldavClient client, String url, String username, String password) {
    run(() -> client.forUrl(url, username, password).getHomeSet());
  }
}

File: app/src/main/java/org/tasks/caldav/UpdateCaldavAccountViewModel.kt
package org.tasks.caldav

import dagger.hilt.android.lifecycle.HiltViewModel
import org.tasks.ui.CompletableViewModel
import javax.inject.Inject

@HiltViewModel
class UpdateCaldavAccountViewModel @Inject constructor(
        private val provider: CaldavClientProvider
) : CompletableViewModel<String>() {
    suspend fun updateCaldavAccount(url: String, username: String, password: String) {
        run { provider.forUrl(url, username, password).homeSet(username, password) }
    }
}